cpp/include/raft/distance/detail/fused_l2_nn.cuh

```diff
@@ -92,14 +92,14 @@ DI void updateReducedVal(
   const auto lid      = threadIdx.x % raft::WarpSize;
   const auto accrowid = threadIdx.x / P::AccThCols;
 
-  // for now have first lane from each warp update a unique output row. This
-  // will resolve hang issues with pre-Volta architectures
+  // Update each output row in order within a warp. This will resolve hang
+  // issues with pre-Volta architectures
 #pragma unroll
   for (int j = 0; j < (raft::WarpSize / P::AccThCols); j++) {
-    if (lid == 0) {
+    if (lid == j * P::AccThCols) {
 #pragma unroll
       for (int i = 0; i < P::AccRowsPerTh; ++i) {
-        auto rid = gridStrideY + accrowid + j + i * P::AccThRows;
+        auto rid = gridStrideY + accrowid + i * P::AccThRows;
```

Comment:
Shouldn't we keep `j` here?

Reply:
No, but it took me a while to figure out why! 

The role of `j` is taken over by `accrowid`, which changes from thread to thread: 
```
const auto accrowid = threadIdx.x / P::AccThCols;
```
Since we are no longer doing all the work in thread 0, we can remove `j`.